tests/trading/test_wcl_fuzzy.py

```diff
@@ -22,100 +22,110 @@
-    completeSets = contractsFixture.contracts['CompleteSets']
+    orders = fixture.contracts['Orders']
+    ordersFetcher = fixture.contracts['OrdersFetcher']
+    createOrder = fixture.contracts['CreateOrder']
+    fillOrder = fixture.contracts['FillOrder']
+    completeSets = fixture.contracts['CompleteSets']
 
     creatorAddress = tester.a1
     fillerAddress = tester.a2
     creatorKey = tester.k1
     fillerKey = tester.k2
 
+    creatorOriginalBalance = fixture.chain.head_state.get_balance(creatorAddress)
+    fillerOriginalBalance = fixture.chain.head_state.get_balance(fillerAddress)
+
     # create order
+    creatorEthRequiredLong = 0 if creatorLongShares == 0 else creatorLongShares * market.getNumTicks()
+    creatorEthRequiredShort = 0 if creatorShortShares == 0 else creatorShortShares * market.getNumTicks()
     acquireLongShares(orderOutcome, creatorLongShares, createOrder.address, sender = creatorKey)
     acquireShortShareSet(orderOutcome, creatorShortShares, createOrder.address, sender = creatorKey)
-    orderID = createOrder.publicCreateOrder(orderType, orderSize, orderPrice, market.address, orderOutcome, longTo32Bytes(0), longTo32Bytes(0), 42, sender = creatorKey, value = creatorTokens)
+    orderId = createOrder.publicCreateOrder(orderType, orderSize, orderPrice, market.address, orderOutcome, longTo32Bytes(0), longTo32Bytes(0), 42, sender = creatorKey, value = creatorTokens)
 
     # validate the order
-    order = ordersFetcher.getOrder(orderID)
+    order = ordersFetcher.getOrder(orderId)
     assert order[ATTOSHARES] == orderSize
     assert order[DISPLAY_PRICE] == orderPrice
     assert order[OWNER] == bytesToHexString(creatorAddress)
     assert order[TOKENS_ESCROWED] == creatorTokens
     assert order[SHARES_ESCROWED] == creatorLongShares or creatorShortShares
 
     # fill order
+    fillerEthRequiredLong = 0 if fillerLongShares == 0 else fillerLongShares * market.getNumTicks()
+    fillerEthRequiredShort = 0 if fillerShortShares == 0 else fillerShortShares * market.getNumTicks()
     acquireLongShares(orderOutcome, fillerLongShares, fillOrder.address, sender = fillerKey)
     acquireShortShareSet(orderOutcome, fillerShortShares, fillOrder.address, sender = fillerKey)
-    remaining = fillOrder.publicFillOrder(orderID, orderSize, 42, sender = fillerKey, value = fillerTokens)
+    remaining = fillOrder.publicFillOrder(orderId, orderSize, 42, sender = fillerKey, value = fillerTokens)
     assert not remaining
 
     # assert final state
-    assert cash.balanceOf(creatorAddress) == expectedMakerTokens
-    assert cash.balanceOf(fillerAddress) == expectedFillerTokens
+    assert fixture.chain.head_state.get_balance(creatorAddress) == creatorOriginalBalance - creatorEthRequiredLong - creatorEthRequiredShort - creatorTokens + expectedMakerTokens
+    assert fixture.chain.head_state.get_balance(fillerAddress) == fillerOriginalBalance - fillerEthRequiredLong - fillerEthRequiredShort - fillerTokens + expectedFillerTokens
```

Comment:
Same comment as on the other PR, this seems incorrect.  The assertion should be that the participants end with the expected amount of ETH.  If `expectedMakerTokens` is a "change in ETH from start" then adding original balance would be reasonable.  The other stuff shouldn't be necessary and suggests that the test has some other problem in it that is causing the balances to not work out correctly.